agent/job_runner.go

```diff
@@ -75,6 +75,8 @@ var ProtectedEnv = map[string]struct{}{
 	"BUILDKITE_SHELL":                    {},
 }
 
+var DefaultAllowedEnv = []string{"^BUILDKITE$", "^BUILDKITE_.*$", "^CI$"}
+
```

Comment:
I first thought about using a protected environment variables list, as filtering out all the environment variables BK set can break the Buildkite agent's behaviour. However, I soon found that the protected list doesn't include all the environment variables that the agent sets and added that default allowed environment variables list there.
I don't think removing the default list is a good idea, but as it can break some workflows, maybe some warnings can also be added to the doc. Something like "not including `"^BUILDKITE$", "^BUILDKITE_.*$", "^CI$"` can break the buildkite agent."
WDYT?